### PR TITLE
Rescue if Token is Invalid

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Add rescue if token is invalid
* Add concern with rescue statement
* Include concern in controllers
* Add controller spec to test for authentication bug

[#113630033]

### CODE REVIEW

**Comment by michaelwehrley** [comment]
Just a side note, b/c this code has to be in the host apps (if they have additional controllers) and in multiple engines, I thought a ruby gem might be a good idea.  I worked on this for bit to see how it would go: https://github.com/michaelwehrley/rescue

**Comment by ericcf** [comment]
Do you want someone to look at this?

**Comment by michaelwehrley** [comment]
Not the gem at this point, but I will submit the PR in a moment.  Thanks!

File: app/models/think_feel_do_engine/concerns/invalid_auth_token.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module ThinkFeelDoEngine
```
Comment (ericcf) [inline app/models/think_feel_do_engine/concerns/invalid_auth_token.rb]: What was the reason for putting this code in `models`? It seems more related to controllers.

File: spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb [hunk 2]
```diff
@@ -41,6 +41,23 @@ module Participants
 
             it { expect(response.status).to eq 400 }
           end
+
+          describe "when authorization token has expired" do
```
Comment (ericcf) [inline spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb]: What about writing a shared example to test the redirect logic? Then you could easily test any controllers that inherited the module.

File: app/models/think_feel_do_engine/concerns/invalid_auth_token.rb (hunk 1 above)
Comment (michaelwehrley) [inline app/models/think_feel_do_engine/concerns/invalid_auth_token.rb]: Good point; it should be in a controllers concerns folder.

File: spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb (hunk 2 above)
Comment (michaelwehrley) [inline spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb]: Great idea!

Comment (michaelwehrley) [inline spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb]: I never knew how helpful `shared_examples` were.  Great recommendation!

File: spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb [hunk 3]
```diff
@@ -42,6 +42,10 @@ module Participants
             it { expect(response.status).to eq 400 }
           end
         end
+
+        it_behaves_like "an untrusted action when auth token has expired" do
```
Comment (ericcf) [inline spec/controllers/think_feel_do_engine/participants/task_status_controller_spec.rb]: Nice. Maybe add this shared example to a passwords controller spec and sessions controller spec?

**Comment by ericcf** [comment]
@michaelwehrley looks good, thanks!

**Pull request merged**